lib/chainlock/chainlock.js

```diff
@@ -0,0 +1,273 @@
+
+/**
+ * @param {Buffer|Object|string} arg - A Buffer, JSON string or Object
+ * @returns {Object} - An object representing chainlock data
+ * @throws {TypeError} - If the argument was not recognized
+ * @private
+ */
+ChainLock._from = function _from(arg) {
+  var info = {};
+  if (BufferUtil.isBuffer(arg)) {
+    info = ChainLock._fromBufferReader(BufferReader(arg));
+  } else if (isObject(arg)) {
+    info = ChainLock._fromObject(arg);
+  } else if (isHexString(arg)) {
+    info = ChainLock.fromHex(arg);
+  } else {
+    throw new TypeError('Unrecognized argument for ChainLock');
+  }
+  return info;
+}
+
+ChainLock._fromObject = function _fromObject(data) {
+  $.checkArgument(data, 'data is required');
+  var blockHash = data.blockHash;
+  var signature = data.signature;
+  if (isString(data.blockHash)) {
+    blockHash = BufferUtil.reverse(Buffer.from(data.blockHash, 'hex'));
+  }
+
+  if (isString(data.signature)) {
+    signature = Buffer.from(data.signature, 'hex');
+  }
+  var info = {
+    height: data.height,
+    blockHash: blockHash,
+    signature: signature
+  }
+  return info;
+}
+
+/**
+ * @param {BufferReader} br - Chainlock data
+ * @returns {Object} - An object representing the chainlock data
+ * @private
+ */
+ChainLock._fromBufferReader = function _fromBufferReader(br) {
+  var info = {};
+  info.height = br.readInt32LE();
+  info.blockHash = br.read(SHA256_HASH_SIZE);
+  info.signature = br.read(BLS_SIGNATURE_SIZE);
+  return info;
+};
+
+/**
+ * @param {BufferReader} br A buffer reader of the block
+ * @returns {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBufferReader = function fromBufferReader(br) {
+  $.checkArgument(br, 'br is required');
+  var data = ChainLock._fromBufferReader(br);
+  return new ChainLock(data);
+};
+
+/**
+ * Creates ChainLock from a hex string.
+ * @param {String} string - A hex string representation of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromHex = ChainLock.fromString = function fromString(string) {
+  return ChainLock.fromBuffer(Buffer.from(string, 'hex'));
+};
+
+
+/**
+ * Creates ChainLock from a Buffer.
+ * @param {Buffer} buffer - A buffer of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBuffer = function fromBuffer(buffer) {
+  return ChainLock.fromBufferReader(new BufferReader(buffer));
+};
+
+/**
+ * Create ChainLock from an object
+ * @param {Object} obj - an object with all properties of chainlock
+ * @return {ChainLock}
+ */
+ChainLock.fromObject = function fromObject(obj) {
+  var data = ChainLock._fromObject(obj);
+  return new ChainLock(data);
+};
+
+/**
+ * Verify that the signature is valid against the Quorum using quorumPublicKey
+ * @param {QuorumEntry} quorumEntry - quorum entry to test signature against
+ * @returns {Boolean} - if the signature is valid for this quorum
+ */
+ChainLock.prototype.verifySignatureAgainstQuorum = function verifySignatureAgainstQuorum(quorumEntry) {
```

Comment:
Maybe we can make it an async function here?

Reply:
Good point ! Fixed + typings and JSDoc